builder/azure/arm/config.go

```diff
@@ -1265,8 +1264,8 @@ func assertManagedImageDataDiskSnapshotName(name, setting string) (bool, error)
 }
 
 func assertResourceNamePrefix(name, setting string) (bool, error) {
-	if !isValidAzureName(reResourceNamePrefix, name) {
-		return false, fmt.Errorf("The setting %s must only contain characters from a-z, A-Z, 0-9 and _ and the maximum length is 10 characters", setting)
+	if !reResourceNamePrefix.Match([]byte(name)) {
```

Comment:
Alternatively here you could use `MatchString`, that wouldn't require converting the name to a byte array

Reply:
Nice catch :) Changed